tests/end-to-end/regression/2155/Issue2155Test.php

```diff
@@ -0,0 +1,34 @@
+<?php declare(strict_types=1);
+/*
+ * This file is part of PHPUnit.
+ *
+ * (c) Sebastian Bergmann <sebastian@phpunit.de>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+namespace PHPUnit\TestFixture\Issue2155;
+
+use function error_log;
+use PHPUnit\Framework\TestCase;
+
+class Foo
+{
+    public function doFoo()
+    {
+        error_log('logged a side effect');
+
+        return '';
+    }
+}
+
+final class Issue2155Test extends TestCase
+{
+    public function testOne(): void
+    {
+        $foo = new Foo;
+
+        $this->assertSame('', $foo->doFoo());
+        self::expectErrorOutputRegex('/logged a side effect/');
```

Comment:
the string written into `error_log()` will get a date format prefix. therefore I have only implemented `expectErrorOutputRegex` as because of the ever changing date format, adding expectatings without a regex will not work.